ldap/servers/slapd/daemon.c

```diff
@@ -1554,43 +1554,54 @@ setup_pr_read_pds(Connection_Table *ct, int listnum)
         if (c->c_state == CONN_STATE_FREE) {
             connection_table_move_connection_out_of_active_list(ct, c);
         } else {
-            /* we try to acquire the connection mutex, if it is already
+            /* Check for a timeout for PAGED RESULTS */
+            if (pagedresults_is_timedout_nolock(c)) {
```

Comment:
I have a doubt here. Without holding the lock can we really trust the value.
I would rather systematically acquire the lock before calling this function.

Reply:
I had a similar concern, but this was Pierre's suggestion so I just went with it :-)  I "think" it's sort of a shortcut to avoid taking a lock at first.  The timer will only go from a value > 0 to 0.  If it's zero then we know its good because it never goes from 0 to > 0.  If it is > 0, then double check with the lock that it hasn't changed to zero.  Is this what you were thinking @progier389 ?

Reply:
FYI: here is the rational for the double check first without lock then with lock:
 1) Timing is critical in this function: as previous issues showed a delay of a few ms leads to a drastic performance drop in term of number of operations p[er seconds
 2) Indeed the check without lock is unreliable but such inconsistency is transient and:
        if we get True instead of False then the second test under lock will catch it. 
        if we get False instead of True then you will try again at next loop (i.e after 250ms in the worst case) as we are simply computing a timeout, it does not really matter.
 3) Most operations are not page search so it would be a pity to impact them by always taking the lock

Reply:
Thanks for the explanation. I agree that it is a perf sensitive loop and avoiding a lock in the general case is good justification. Also the timeout will be eventually detected.

Reply:
> Sorry, I have another doubt.. (do you know columbo ;)

_"Oh just one more thing..."_

Reply:
Talked with Thierry offline.  We will look into this issue in a new ticket.